File: ce/customer-engagement/on-premises/snippets/csharp/CRMV8/businessmanagement/cs/associatereciprocalconnectionroles.cs
// <snippetassociatereciprocalconnectionroles>


 // Associate the connection roles
  AssociateRequest associateConnectionRoles = new AssociateRequest
 {
     Target = new EntityReference(ConnectionRole.EntityLogicalName,
         _connectionRole1Id),
     RelatedEntities = new EntityReferenceCollection()
     {
         new EntityReference(ConnectionRole.EntityLogicalName,
             _connectionRole2Id)
     },
     // The name of the relationship connection role association 
     // relationship in MS CRM
     Relationship = new Relationship()
     {
         PrimaryEntityRole = EntityRole.Referencing, // Referencing or Referenced based on N:1 or 1:N reflexive relationship.
         SchemaName = "connectionroleassociation_association"
     }
 };

 _serviceProxy.Execute(associateConnectionRoles);
 Console.WriteLine("Associated the connection roles.");

// </snippetassociatereciprocalconnectionroles>

File: ce/customerengagement/on-premises/snippets/visualbasic/CRMV8/businessmanagementvb/vb/associatereciprocalconnectionroles.vb
' <snippetassociatereciprocalconnectionroles>


 ' Associate the connection roles
 Dim associateConnectionRoles As AssociateRequest =
     New AssociateRequest With
     {
         .Target = New EntityReference(ConnectionRole.EntityLogicalName,
                                       _connectionRole1Id),
         .RelatedEntities =
         New EntityReferenceCollection() From
         {
             New EntityReference(ConnectionRole.EntityLogicalName,
                                 _connectionRole2Id)
         },
         .Relationship =
         New Relationship() With
         {
             .PrimaryEntityRole = EntityRole.Referencing,
             .SchemaName = "connectionroleassociation_association"
         }
     }
 ' The name of the relationship connection role association 
 ' relationship in MS CRM

 _serviceProxy.Execute(associateConnectionRoles)
 Console.WriteLine("Associated the connection roles.")

' </snippetassociatereciprocalconnectionroles>